### ISSUE TITLE
IS-04-01 test_12 not discovering all IP addresses?

### ISSUE THREAD

**Issue by garethsb-sony** [AMWA-TV/nmos-testing#214]
From https://github.com/AMWA-TV/nmos-testing/pull/213#issuecomment-486663398:

Another issue @simonlo-sony has seen with `test_12` is that `MdnsListener.get_service_list` only ever seems to return one entry, even when the test suite and the node-under-test are run on the same machine with multiple interfaces. `avahi-browse` (or `dns-sd`) correctly resolves multiple addresses in this case, but `MdnsListener.get_service_list` only returns one. Sometimes it _is_ the primary address and the test passes, other times it gets back the 127.0.0.1 advertisement and the test fails. I can't actually see how python-zeroconf module can return more than one address... :-/

**Comment by andrewbonney** [comment]
I'm wondering if the simplest way to resolve this may be to limit zeroconf browsing to a single interface. This would match the default/preferred interface. If you wanted to test a multi-interface Node properly you'd need to run the tests twice anyway by entering different IPs into the UI. This would then allow you to match the entered IP to the configured interface (with the help of https://github.com/AMWA-TV/nmos-testing/pull/231)

**Comment by garethsb-sony** [comment]
We now have `BIND_INTERFACE` but it's not wired up to zeroconf - actually, it's not clear to me that [python-zeroconf](https://github.com/jstasiak/python-zeroconf/) supports specifying an interface for browsing yet. https://github.com/jstasiak/python-zeroconf/pull/104 looks like it tries to address this.

**Comment by garethsb-sony** [comment]
@maweit has identified this also affects `IS0403Test.test_01`.

**Comment by garethsb-sony** [comment]
Hmm, is https://github.com/jstasiak/python-zeroconf/issues/193 the fix we need? (Line numbers in the suggested patch don't correspond, but it's obvious enough what's intended.) Will try this out.

**Comment by garethsb-sony** [comment]
Works for me, after applying the above patch, and using `netifaces.addresses` to map `BIND_INTERFACE` to a local IP address and constructing `Zeroconf` with that address. Will see if they merge the fix.

**Comment by maweit** [comment]
Was not successful IS0403Test.test_01. Still failing on a two interface machine. Maybe my merge wasn't correct.

**Comment by garethsb-sony** [comment]
@maweit, did you modify the `IS0403Test.set_up_tests` to pass the IP address to bind to the `Zeroconf` constructor?

**Comment by rhastie** [comment]
+1 for the issue...

I've just found that the Testing tool incorrectly fails the IS-04-02 mDNS advertisement tests (test_01 and test_02) when the docker container has more than one interface. In my scenario the testing tool was receiving valid register/query advertisements on both interfaces during the test.

For the time being I can work round this problem by just using a single interface on the container but I suspect this will affect others...

**Comment by garethsb-sony** [comment]
Yes, we've used various workarounds, disabling all other interfaces, or just reconfiguring the avahi-daemon like so: ([link](https://github.com/sony/nmos-cpp/commit/e20ba421b74095c407f6cbfa538dea32a6252995))

```sh
sudo sed -i 's/#*allow-interfaces=.*/allow-interfaces=eth0/g' /etc/avahi/avahi-daemon.conf
sudo systemctl restart avahi-daemon
```

It'd be nice to fix for sure. We may be able to wire up `BIND_INTERFACE` with zeroconf 0.28.0+.
